hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/ReconfigOMSubcommand.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.net.NetUtils;
+import org.apache.hadoop.ozone.om.helpers.OMNodeDetails;
+import org.apache.hadoop.ozone.om.protocolPB.OMAdminProtocolClientSideImpl;
+import org.apache.hadoop.security.UserGroupInformation;
+import picocli.CommandLine;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+/**
+ * Handler of om roles command.
+ */
+@CommandLine.Command(
+    name = "reconfig",
+    customSynopsis = "ozone admin om reconfig -address=<ip:port> -op=" +
+        "start|status|properties",
+    description = "Dynamic reconfiguring without stopping the OM.\n" +
+        "Three operations are provided:\n" +
+        "\tstart:      Execute the reconfig operation asynchronously\n" +
+        "\tstatus:     Check reconfiguration status\n" +
+        "\tproperties: List reconfigurable properties",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class ReconfigOMSubcommand implements Callable<Void> {
+
+  @CommandLine.ParentCommand
+  private OMAdmin parent;
+
+  @CommandLine.Option(names = {"-address", "--host-port"},
+      description = "om address: <ip:port> or <hostname:port>",
+      required = true)
+  private String address;
+
+  @CommandLine.Option(names = {"-op"},
+      description = "op",
+      required = true)
+  private String op;
+
+  private OzoneConfiguration ozoneConf;
+  private UserGroupInformation user;
+
+  @Override
+  public Void call() throws IOException {
+    ozoneConf = new OzoneConfiguration();
+    user = UserGroupInformation.getCurrentUser();
+
+    InetSocketAddress socketAddr = NetUtils.createSocketAddr(address);
+    OMNodeDetails omNode = new OMNodeDetails.Builder().setRpcAddress(socketAddr)
+        .build();
+    try (OMAdminProtocolClientSideImpl omAdminProtocolClient = 
+             OMAdminProtocolClientSideImpl.createProxyForSingleOM(ozoneConf,
+                 user, omNode)) {
+
+      if ("start".equals(op)) {
+        omAdminProtocolClient.startOmReconfiguration();
+        System.out.printf("Started OM reconfiguration task on node [%s].%n",
+            address);
+
+      } else if ("status".equals(op)) {
+        ReconfigurationTaskStatus status =
+            omAdminProtocolClient.getOmReconfigurationStatus();
+        System.out.printf("Reconfiguring status for node [%s]: ", address);
+        printReconfigurationStatus(status);
+
+      } else if ("properties".equals(op)) {
+        List<String> properties =
+            omAdminProtocolClient.listOmReconfigurableProperties();
+        System.out.printf("OM Node [%s] Reconfigurable properties:%n", address);
+        for (String name : properties) {
+          System.out.println(name);
+        }
+      } else {
+        System.err.println("Unknown operation: " + op);
+      }
```

Comment:
Please use picocli builtin support for separate subcommands instead of implementing a custom way to specify operations.  See `SafeModeCommands` for example.